### PR TITLE
PSAP-1003: NTO: reapply_sysctl and verify Profiles improvements

### PR DESCRIPTION
**Pull request by jmencak**
This PR adds an example of using reapply_sysctl to override host-level sysctls and adds information on checking about the NTO and its node agents' health.

Version(s):
4.13

Issue:
https://issues.redhat.com/browse/PSAP-1003

Link to docs preview:
https://56773--docspreview.netlify.app/openshift-enterprise/latest/scalability_and_performance/using-node-tuning-operator.html

QE review:
- [ ] QE has approved this change.

### CODE REVIEW

File: modules/custom-tuning-example.adoc [hunk 1]
```diff
@@ -84,3 +84,45 @@ the `openshift-node` TuneD daemon profile shipped within the default
...
+
+*Overriding host-level sysctls*
+
+Various kernel parameters can be changed at runtime via `/run/sysctl.d/`,
+`/etc/sysctl.d/` and `/etc/sysctl.conf` host configuration files.
+The {product-title} itself adds several host configuration files
+which set kernel parameters at runtime. Most notably `net.ipv[4-6].`,
+`fs.inotify.` and `vm.max_map_count`. These runtime parameters provide a
+very basic functionality tuning for the system prior to the kubelet and the
+Operator start.
```
Comment (dagrayvid) [inline modules/custom-tuning-example.adoc:96]:
```suggestion
`fs.inotify.` and `vm.max_map_count`. These runtime parameters provide
very basic functional tuning for the system prior to the kubelet and the
Operator start.
```

File: modules/cluster-node-tuning-operator-verify-profiles.adoc [hunk 2]
```diff
@@ -28,3 +28,22 @@ worker-b         openshift-node            True      False      6h28m
...
+----
+
+.Example output
+[source,terminal]
+----
+NAME          VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+node-tuning   4.13.1    True        False         True       60m     1/5 Profiles with bootcmdline conflict
+----
+
+If either the ClusterOperator/node-tuning or a Profile object's status
```
Comment (sheriff-rh) [inline modules/cluster-node-tuning-operator-verify-profiles.adoc:47]: Can you remove the hard wrap? Our [guidelines](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#general-file-guidelines) recommend undoing it where possible.

File: modules/cluster-node-tuning-operator-verify-profiles.adoc [hunk 3]
```diff
@@ -28,3 +28,22 @@ worker-b         openshift-node            True      False      6h28m
...
+is Degraded, additional information will be provided in the operator or
```
Comment (sheriff-rh) [inline modules/cluster-node-tuning-operator-verify-profiles.adoc:48]: Let's change this to `DEGRADED` in backticks to match line 29 above.

File: modules/custom-tuning-example.adoc [hunk 4]
```diff
@@ -84,3 +84,45 @@ the `openshift-node` TuneD daemon profile shipped within the default
...
+Various kernel parameters can be changed at runtime via `/run/sysctl.d/`,
+`/etc/sysctl.d/` and `/etc/sysctl.conf` host configuration files.
+The {product-title} itself adds several host configuration files
+which set kernel parameters at runtime. Most notably `net.ipv[4-6].`,
+`fs.inotify.` and `vm.max_map_count`. These runtime parameters provide
+very basic functional tuning for the system prior to the kubelet and the
+Operator start.
+
+The Operator respects these settings and will not override them, unless the
+`reapply_sysctl` option is set to `false`.  This will result in `TuneD` not
```
Comment (sheriff-rh) [inline modules/custom-tuning-example.adoc:99]: There's two spaces after the period - delete one space.

File: modules/custom-tuning-example.adoc [hunk 5]
```diff
@@ -84,3 +84,45 @@ the `openshift-node` TuneD daemon profile shipped within the default
 Tuned CR to use OpenShift-specific tuning for compute nodes.
 
 // Note the issues with including profiles sharing the same ancestor: see link:https://bugzilla.redhat.com/show_bug.cgi?id=1825882[BZ#1825882]
+
+*Overriding host-level sysctls*
+
+Various kernel parameters can be changed at runtime via `/run/sysctl.d/`,
```
Comment (sheriff-rh) [inline modules/custom-tuning-example.adoc:90]: Remove hard wrapping for this text as well please.

File: modules/cluster-node-tuning-operator-verify-profiles.adoc [hunk 6]
```diff
@@ -28,3 +28,22 @@ worker-b         openshift-node            True      False      6h28m
 * `APPLIED`: `True` if the TuneD daemon applied the desired profile. (`True/False/Unknown`).
 * `DEGRADED`: `True` if any errors were reported during application of the TuneD profile (`True/False/Unknown`).
 * `AGE`: Time elapsed since the creation of Profile object.
+
+The ClusterOperator/node-tuning object also contains useful information about the Operator and its node agents' health.
+For example, operator misconfiguration will be reported via ClusterOperator/node-tuning status messages.
```
Comment (sheriff-rh) [inline modules/cluster-node-tuning-operator-verify-profiles.adoc:33]: There's a couple places where you've written "operator" and "Operator". Guidelines require that it be written capitalized Operator throughout.

**Review by sheriff-rh — COMMENTED** [review]
Looks great, Jiri! I have some small requests. 

I also assigned this pull request to myself to keep track of it. After you make these changes, I'll put the PR through our peer review process, which may require more adjustments. Lastly, we will verify that all changes are squashed into one commit, and then I can merge the content. Please let me know if you have any questions!

File: modules/custom-tuning-example.adoc (hunk 4 above)
Comment (jmencak) [reply modules/custom-tuning-example.adoc:99]: Cannot teach an old dog new tricks.  Always do two spaces after an end of a sentence.  Removed here as requested then.

File: modules/cluster-node-tuning-operator-verify-profiles.adoc (hunk 6 above)
Comment (jmencak) [reply modules/cluster-node-tuning-operator-verify-profiles.adoc:33]: Ack.  And removed a hard-wrap here too.

File: modules/cluster-node-tuning-operator-verify-profiles.adoc [hunk 7]
```diff
@@ -28,3 +28,19 @@ worker-b         openshift-node            True      False      6h28m
 * `APPLIED`: `True` if the TuneD daemon applied the desired profile. (`True/False/Unknown`).
 * `DEGRADED`: `True` if any errors were reported during application of the TuneD profile (`True/False/Unknown`).
 * `AGE`: Time elapsed since the creation of Profile object.
+
+The ClusterOperator/node-tuning object also contains useful information about the Operator and its node agents' health. For example, Operator misconfiguration will be reported via ClusterOperator/node-tuning status messages.
```
Comment (kcarmichael08) [inline modules/cluster-node-tuning-operator-verify-profiles.adoc:32]:
```suggestion
The `ClusterOperator/node-tuning` object also contains useful information about the Operator and its node agents' health. For example, Operator misconfiguration is reported by `ClusterOperator/node-tuning` status messages.
```

File: modules/cluster-node-tuning-operator-verify-profiles.adoc [hunk 8]
```diff
@@ -28,3 +28,19 @@ worker-b         openshift-node            True      False      6h28m
...
+
```
Comment (kcarmichael08) [inline modules/cluster-node-tuning-operator-verify-profiles.adoc:33]: Needs an introductory sentence before the code block, such as "To get status information about the `ClusterOperator/node-tuning` object, run the following command:"

File: modules/cluster-node-tuning-operator-verify-profiles.adoc [hunk 9]
```diff
@@ -28,3 +28,19 @@ worker-b         openshift-node            True      False      6h28m
...
+----
+
+.Example output
+[source,terminal]
+----
+NAME          VERSION   AVAILABLE   PROGRESSING   DEGRADED   SINCE   MESSAGE
+node-tuning   4.13.1    True        False         True       60m     1/5 Profiles with bootcmdline conflict
+----
+
+If either the ClusterOperator/node-tuning or a Profile object's status is `DEGRADED`, additional information will be provided in the operator or operand logs.
```
Comment (kcarmichael08) [inline modules/cluster-node-tuning-operator-verify-profiles.adoc:46]:
```suggestion
If either the `ClusterOperator/node-tuning` or a profile object's status is `DEGRADED`, additional information is provided in the Operator or operand logs.
```

File: modules/custom-tuning-example.adoc [hunk 10]
```diff
@@ -84,3 +84,36 @@ the `openshift-node` TuneD daemon profile shipped within the default
 Tuned CR to use OpenShift-specific tuning for compute nodes.
 
 // Note the issues with including profiles sharing the same ancestor: see link:https://bugzilla.redhat.com/show_bug.cgi?id=1825882[BZ#1825882]
+
+*Overriding host-level sysctls*
+
+Various kernel parameters can be changed at runtime via `/run/sysctl.d/`, `/etc/sysctl.d/` and `/etc/sysctl.conf` host configuration files.  The {product-title} itself adds several host configuration files which set kernel parameters at runtime. Most notably `net.ipv[4-6].`, `fs.inotify.` and `vm.max_map_count`. These runtime parameters provide very basic functional tuning for the system prior to the kubelet and the Operator start.
```
Comment (kcarmichael08) [inline modules/custom-tuning-example.adoc:90]:
```suggestion
Various kernel parameters can be changed at runtime by using `/run/sysctl.d/`, `/etc/sysctl.d/`, and `/etc/sysctl.conf` host configuration files. {product-title} adds several host configuration files which set kernel parameters at runtime; for example, `net.ipv[4-6].`, `fs.inotify.`, and `vm.max_map_count`. These runtime parameters provide basic functional tuning for the system prior to the kubelet and the Operator start.
```

File: modules/custom-tuning-example.adoc [hunk 11]
```diff
@@ -84,3 +84,36 @@ the `openshift-node` TuneD daemon profile shipped within the default
...
+
+The Operator respects these settings and will not override them, unless the `reapply_sysctl` option is set to `false`. This will result in `TuneD` not applying the settings from the host configuration files after it applies its custom profile.
```
Comment (kcarmichael08) [inline modules/custom-tuning-example.adoc:92]:
```suggestion
The Operator does not override these settings unless the `reapply_sysctl` option is set to `false`. Setting this option to `false` results in `TuneD` not applying the settings from the host configuration files after it applies its custom profile.
```

Comment (kcarmichael08) [reply modules/custom-tuning-example.adoc:92]: I think my suggestion is accurate but I wasn't 100% sure what "this" referred to - setting the option to false, or overriding the settings, or something else.

**Review by kcarmichael08 — COMMENTED** [review]
I had a few suggestions for wording per IBM/RH Style but otherwise everything lgtm!

File: modules/custom-tuning-example.adoc (hunk 11 above)
Comment (jmencak) [reply modules/custom-tuning-example.adoc:92]: Your suggestion is accurate.  Thank you.  Accepted.

**Comment by jmencak** [comment]
Thank you for the reviews, all.  I've accepted all suggestions and squashed the commits.  Hopefully the PR is ready now.

**Review by sheriff-rh — APPROVED** [review]
LGTM!

**Comment by sheriff-rh** [comment]
All approvals in place. Merging!

**Pull request merged**